lib/os-cluster-entrypoint.ts

```diff
@@ -86,6 +86,12 @@ export class OsClusterEntrypoint {
       }
       const security = securityDisabled === 'true';
 
+      // adminPassword is required if security is enabled and demo config is to be run for versions 2.12 and above
+      const adminPassword: String = security ? `${scope.node.tryGetContext('adminPassword')}` : "";    
+      if (!security && Number.parseFloat(distVersion) >= 2.12 && (adminPassword === null || adminPassword === "")) {
```

Comment:
` (adminPassword === null || adminPassword === "")` could just be `adminPassword === 'undefined'`